src/Symfony/Component/EventDispatcher/EventListenerTrait.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+use Symfony\Contracts\EventDispatcher\Event as ContractsEvent;
+
+/**
+ * Helper trait for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+trait EventListenerTrait
+{
+    /**
+     * {@inheritdoc}
+     */
+    final public static function getSubscribedEvents(): array
+    {
+        return [static::getEventName() => [static::getMethodName(), static::getPriority()]];
+    }
+
+    public static function getEventName(): string
+    {
+        $method = new \ReflectionMethod(static::class, static::getMethodName());
+
+        if (0 === $method->getNumberOfParameters()) {
+            throw new \LogicException(sprintf('We cannot infer the Event name from "%s::%s()" method without parameters, you must to define one or implement the "%s::getEventName()" method to return it explicitly.', static::class, $method->getName(), static::class));
+        }
+
+        $parameter = $method->getParameters()[0];
+
+        if (null === $type = $parameter->getType()) {
+            throw new \LogicException(sprintf('We cannot infer the Event name from "%s::%s()" method without type-hint the "$%s" parameter, you must to type-hint it or implement the "%s::getEventName()" method to return it explicitly.', ContractsEvent::class, $method->getName(), $parameter->getName(), static::class));
+        }
+
+        if ($type->isBuiltin()) {
+            throw new \LogicException(sprintf('The "$%s" parameter defined in "%s::%s()" method must be a class of "%s", "%s" given.', $parameter->getName(), static::class, $method->getName(), ContractsEvent::class, $type->getName()));
+        }
+
+        return $type->getName();
+    }
+
+    public static function getMethodName(): string
+    {
+        return '__invoke';
```

Comment:
I would suggest to force only one method name: `__invoke`. The trait should not become and extensibility unit to me.